bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagJournal.kt

```diff
@@ -194,9 +194,24 @@ internal class BugsnagJournal @JvmOverloads internal constructor(
          * - All occurrences of special characters (+ . \)
          */
         fun unspecialMapPath(pathComponent: String): String {
-            val index = pathComponent.toIntOrNull()
-            if (index != null) {
-                return "\\" + pathComponent
+            if (pathComponent.length == 0) {
+                return pathComponent
+            }
+            val firstChar = pathComponent[0]
+            if (firstChar == '-' || (firstChar >= '0' && firstChar <= '9')) {
+                val index = pathComponent.toIntOrNull()
+                if (index != null) {
+                    return "\\" + pathComponent
+                }
+            }
+
+            if (!(
+                pathComponent.contains('.') ||
+                    pathComponent.contains('+') ||
+                    pathComponent.contains('\\')
```

Comment:
```suggestion
                pathComponent.indexOfFirst { it == '.' || it == '+' || it == '\\' } != -1
```
then we only have a single string scan?